plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java

```diff
@@ -75,6 +75,7 @@ private AppEngineRuntimeInstallJob(IFacetedProject facetedProject) {
       super(Messages.getString("appengine.install.runtime.to.project", // $NON-NLS$
           facetedProject.getProject().getName()));
       this.facetedProject = facetedProject;
+      setRule(facetedProject.getProject());
```

Comment:
I'm concerned about possible deadlock: this job waits for the first `ConvertJob`, and when this job is being created (or scheduled), the first `ConvertJob` is already scheduled, which may be in the sleeping state. If we grab a lock, the `ConvertJob` may not begin.